### DISCUSSION TITLE
next/image fallback image.

### DISCUSSION THREAD

**Discussion by bobbyflowstate** [vercel/next.js#19544] [Help]
Hi there, I'm just starting to use Image from next/image in my project and I need to implement a fallback image.

For example, with user avatars I need a fallback if they haven't uploaded one. I've looked around and I don't see any mention of a way to do this. Your help is much appreciated!

**Comment by doptster** [comment]
I don't think a fallback image is needed in this scenario. You can just simply switch the `src` of the image depending on the user upload status.

**Comment by bobbyflowstate** [comment]
Hey actually my example was a poor example. I'm dealing with situations where the image is broken. For example I'm loading images from a third party API and while the image URL exists, when I load them, some of them are broken. In these cases I need a fallback image.

**Comment by bobbyflowstate** [comment]
Actually this is how I solved it. Maybe this is obvious, but leaving it here for those who were unaware of this solution:

```js
const ImageWithFallback: React.FC<ImageWithFallbackProps> = ({ imgUrl, fallbackUrl }) => {
    const [imgError, setImgError] = useState(false);

    return (
        <>
            {!imgError &&
                <img src={imgUrl} onError={(ev) => {
                    setImgError(true);
                }}/>}
            {!!imgError &&
                <img src={fallbackUrl}/>}
        </>
    )
}
```

**Comment by clevermiraz** [comment]
```js
const ImageWithFallback = ({ fallback = fallbackImage, alt, src, ...props }) => {
    const [error, setError] = useState(null);

    useEffect(() => {
        setError(null);
    }, [src]);

    return <Image alt={alt} onError={setError} src={error ? fallbackImage : src} {...props} />;
};
```

refer: https://dev.to/frontenddeveli/configure-fallback-images-in-react-and-nextjs-54ej
